### ISSUE TITLE
run different CA in different normal world OS (based on hypervisor)

### ISSUE THREAD

**Issue by cxdinter** [OP-TEE/optee_os#2249]
Hi ,
  If there is a hardware platform (ARMv8) which running two guest OS (AGL and Android) based on hypervisor, and op-tee os running on Turst Zone. Is it possible to run different CA in each guest os ? Any conflict between two guest os CA when they communicate with op-tee TA concurrently? 

  Please give me some guide, thank you.

**Comment by jenswi-linaro** [comment]
The short answer is that virtualization isn't supported with OP-TEE, there's still a few problems to sort out.
@lorc is doing some work with regards to virtualization and OP-TEE.

**Comment by lorc** [comment]
Yes. I have a sort of PoC that is able to support calls from two guests simultaneously. Right now I'm working on proper implementation but it requires changes in the OP-TEE core, so work there is going slowly.

**Comment by lorc** [comment]
Yes. Actually my company is aiming into automotive sector. And we really need to make OP-TEE virtualization-aware for our projects. This will be open-sourced of course.

**Comment by cxdinter** [comment]
@lorc 
Excuse me, could you please share what changes should be done in opt-tee core for virtualization?
Sorry, I am not familiar with op-tee core, just want to understand the real mechanism of op-tee virtualization.

**Comment by jbech-linaro** [comment]
@cxdinter , there is a new PR (#2370) from @lorc , maybe that is a good starting point.
